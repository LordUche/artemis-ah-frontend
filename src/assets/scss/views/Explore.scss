@import "./base.scss";
.explore {
  &_header {
    width: 100%;
    height: 360px;
    background-color: #333333;
    position: relative;

    &__img {
      width: 100%;
      height: 250px;
      position: absolute;
      top: 0;
      z-index: 0;
    }

    &__text {
      text-align: center;
      font-size: 30px;
      padding-top: 4%;
      color: #ffffff;
      position: absolute;
      z-index: 1;
      left: 0;
      right: 0;
      margin-left: auto;
      margin-right: auto;
    }

    &__input {
      width: 60%;
      height: 60px;
      background: #eeeeee;
      border: none;
      border-radius: 5px;
      font-size: 15px;
      padding: 10px 10px 10px 60px;
      font-family: inherit;
      left: 0;
      right: 0;
      margin-top: 8%;
      margin-left: auto;
      margin-right: auto;
      position: absolute;
    }

    &_filter {
      position: absolute;
      margin-bottom: 1%;
      bottom: 0;
      width: 55%;
      left: 0;
      right: 0;
      margin-left: auto;
      margin-right: auto;
      text-align: center;
      &__fields {
        width: 50%;
        float: left;
      }
    }
  }

  &_body {
    width: 60%;
    margin: 5% auto;
    margin-bottom: 2rem;
  }

  &_hide {
    display: none;
  }
}

#explore_nav {
  box-shadow: none;
}

#explore_input {
  background-image: url("../img/search.png");
  background-repeat: no-repeat;
  background-position: 20px center;
  font-size: 25px;
}
.explore_label {
  color: #ffffff;
  padding-right: 5%;
  font-size: 20px;
}

.explore_select {
  width: 50%;
  display: inline-block;
}

.explore_select_input__control {
  border-radius: 0 !important;
}

.explore_select_input__value-container div {
  color: white;
}

.explore_select_input__indicator-separator {
  background-color: #333333 !important;
}

@media only screen and (max-width: 1024px) {
  /* For mobile phones: */
  .explore_header {
    height: 400px;

    &__img {
      height: 250px;
    }
    &__input {
      width: 60%;
      margin-top: 11%;
    }
    &_filter {
      width: 60%;
      &__fields {
        width: 100%;
        margin-bottom: 1.5%;
      }
    }
  }
}

@media only screen and (max-width: 768px) {
  /* For mobile phones: */
  .explore_header {
    &__img {
      height: 240px;
    }

    &__input {
      width: 80%;
      margin-top: 13%;
    }
  }
  .explore_body {
    width: 80%;
  }
}

@media only screen and (max-width: 600px) {
  /* For mobile phones: */
  .explore_header {
    &__input {
      margin-top: 16%;
    }
  }
}

@media only screen and (max-width: 425px) {
  /* For mobile phones: */
  .explore_header {
    height: 350px;

    &__img {
      height: 200px;
    }

    &__input {
      width: 90%;
      margin-top: 20%;
    }

    &_filter {
      width: 80%;
    }

    &__text {
      padding-top: 6%;
      font-size: 25px;
    }
  }
}

@media only screen and (max-width: 320px) {
  /* For mobile phones: */
  .explore_header {
    &__input {
      margin-top: 23%;
    }
    &__text {
      padding-top: 8%;
    }
  }
}