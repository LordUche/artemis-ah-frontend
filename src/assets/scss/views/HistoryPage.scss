@import "./base.scss";

.history_item{
  cursor: pointer;
  position: relative;
  background: $white-color;
  border-radius: 5px;
  min-height: 100px;
  padding: 70px 20px 20px;
  font-size: 16px;
  box-shadow: 2px 1px 10px rgba(0, 0, 0, 0.25);
  margin: 40px 0;

  &:nth-of-type(odd){
    background: #E4EDFF;
  }

  .history_article_title{
    position: absolute;
    top: 20px;
    left: -20px;
    background: $primary-color;
    color: $white-color;
    padding: 5px 5vw;
    box-shadow: 2px 1px 10px rgba(0, 0, 0, 0.25);
    border-radius: 5px;
  }

  .history_time{
    text-align: right;
  }

  .history_article_details{
    font-size: 14px;

    p:first-of-type{
      color: #000;
      margin-bottom: 10px;
    }

    p:last-of-type{
      color: #4B4B4B;
    }
  }
}