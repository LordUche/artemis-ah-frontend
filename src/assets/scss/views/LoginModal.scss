@import "../base.scss";

.login_body{
    padding: 0;
    color: #656464;

    &_required_text{
      color: $red-color;
      text-align: center;
      font-size: 15px;
    }

    &_info{
      display: flex;
      padding: 40px 30px 70px;
      width: 750px;
    }

    &_socials{
      width: 42.5%;
      padding: 10px 2%;

      &_btn{
        width: 100%;
        color: $white-color;
        margin: 20px 0;
        border: none;

        &_img{
          transform: scale(1.3);
          margin-left: -30px;
          margin-right: 20px !important;
        }
      }

      &_facebook{
        background: #385592;
      }

      &_twitter{
        background: #48A2D7;

        img{
            margin-bottom: -2px;
        }
      }

      &_google{
        background: #C53F2E;

        img{
            margin-bottom: -3px;
        }
      }

      &_responsive{
        display: none;
        width: 0;
      }
    }
    
    &_divider{
      width: 10%;
      position: relative;

      &_line{
        transform: rotate(90deg);
        border: #BCBCBC 1px solid;
        width: 350px;
        position: absolute;
        top: 50%;
        left: -135px;
        right: 0;
        margin: 0 auto;
        z-index: 1010;
      }

      &_span{
        font-size: 20px;
        font-weight: bold;
        border: 1px solid ;
        border-radius: 50%;
        padding: 10px;
        position: absolute;
        top: 30%;
        left: 10px;
        right: 0;
        margin: 0 auto;
        width: fit-content;
        z-index: 1011;
        background: $white-color;
      }
    }

    &_form{
      width: 47.5%;
      padding: 14px 2%;
      
      &_input{
        width: calc(100% - 20px);
        margin: 15px 0 28px;
      }

      &_rem_div{
        width: 100%;
        display: flex;
        margin: -12px 10px 20px;

        &_input{
          width: 20px;
          height: 20px;
          margin-right: 7px;
        }

        &_text{
          font-size: 14px;
        }
      }

      &_error{
        color: $red-color;
        text-align: center;
        margin: -25px 0 25px;

        &_name{
          margin-bottom: 10px;
        }

        &_general{
          color: $white-color;
          background: $red-color;
          font-size: 15px;
          text-transform: capitalize;
          width: calc(100% - 20px);  
          margin: -30px auto 0;
          border-radius: 5px;
          padding: 5px 0;
          text-align: center;
        }
      }

      &_submit_btn{
        width: calc(100% - 20px);
        background: $primary-color;
        color: $white-color;
        border: none;
        margin: 0 0 20px 10px;
      }

      &_links{
        justify-content: space-between;
        width: calc(100% - 20px);
        margin-left: 10px;
        display: flex;
      }
      &_link{
        color: $primary-color;
      }
    }

    &_instruction {
      text-align: center;
      margin: 20px 0;
      font-size: 16px;
      font-weight: 700;
      color: $primary-color;
    }

    &_reset_section {
      text-align: center;
    }

    &_reset_input {
      font-size: 1.6rem;
      font-family: inherit;
      border: 1px solid #cccccc;
      background: #f7f7f7;
      width: 60%;
      margin-top: 4rem;
    }

    &_reset_btn {
      width: 35%;
      margin: 40px 0 20px 0;
      outline: none;
    }

    &_response_text {
      margin: 20px 0;
      color: #000;
      text-transform: capitalize;      
    }
}

#login-register-btn{
  padding: 0;
  font-size: 1.6rem;
}

@media screen and (max-width: 860px){
    .login_body{
        width: 100%;
    
        &_required_text{
          text-align: center;
          font-size: 13px;
        }
    
        &_info{
          width: 100%;
          display: flex;
          flex-wrap: wrap;
          padding: 10px 0px 0px;
        }
    
        &_socials{
          width: 0;
          display: none;

          &_responsive{
            width: 100%;
            order:3;
            display: flex;
            flex-wrap: wrap;

            &_text{
                color: #656464;
                text-align: center;
                font-size: 20px;
                width: 100%;
                margin: 30px 0 20px;
            }

            &_btn{
                &_img{
                  transform: scale(2);
                  margin: 5px !important;
                }
            }

            button{
              width:fit-content;
              margin: 10px auto;
            }

            button:first-of-type{
              margin-left: 10%;
            }

            button:last-of-type{
                margin-right: 10%;
              }
          }
        }
        
        &_divider{
          order: 2;
          width: 100%;
          height: 50px;
    
          &_line{
            transform: rotate(0);
            width: calc(100% + 60px);
            top: 75%;
            left: -30px;
          }
        }
    
        &_form{
          width: 100%;
          order: 1;
          padding: 0px 2%;

          &_input{
            margin: 5px 0;
          }

          &_rem_div{
            margin: 5px 10px 10px;
          }

          &_error{
            margin: 0;
          }
        }
    }
}

@media screen and (max-width: 500px){
    .login_body{
        &_modal{
          padding: 30px 5px;
          top: -10%;
        }

        &_form{
            &_link{
              font-size: 13px;
            }
        }

        &_divider{
      
            &_line{
              width: calc(100% + 10px);
              left: -5px;
            }
          }

        &_reset_input {
          width: 85%;
        }

        &_reset_btn {
          width: 90%;
        }
    }
}