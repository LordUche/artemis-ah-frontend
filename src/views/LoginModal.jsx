import React, { Component } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import {
  bool, func, objectOf, object
} from 'prop-types';
import { Redirect } from 'react-router-dom';
// import { post } from 'axios';
import { post } from 'axios';

// Social Media Auth URL's
import BASE_URL, {
  GOOGLE_SOCIAL_LOGIN_URL,
  FACEBOOK_SOCIAL_LOGIN_URL,
  TWITTER_SOCIAL_LOGIN_URL,
} from '../redux/actions';

// Components
import Modal from '../components/Modal';
import Button from '../components/Button';
import InputField from '../components/InputField';

// Images
import faceBookLogo from '../assets/img/facebooklogo.svg';
import twitterLogo from '../assets/img/twitterlogo.svg';
import googleLogo from '../assets/img/googlelogo.svg';

// Actions
import {
  loginUserAction,
  loadingAuthAction,
  clearAuthErrorAction
} from '../redux/actions/authActions';

/**
 * @description loginmodal
 * @param {object} e event object
 * @returns {HTMLDivElement} login modal
 */
export class LoginModal extends Component {
  state = {
    name: '',
    password: '',
    rememberMe: false,
    resetPasswordModal: false,
    email: '',
    message: '',
    fetching: false
  };

  clearErrors = () => {
    const { clearAuthError, errorMessages } = this.props;
    if (errorMessages.errors) {
      clearAuthError();
    }
  };

  componentWillMount = () => {
    this.clearErrors();
  };

  handleInputChange = (e) => {
    this.clearErrors();
    this.setState({
      [e.target.name]: e.target.value
    });
  };

  handleSubmit = (e) => {
    e.preventDefault();
    const { loginUser, loadingAuth } = this.props;
    loadingAuth();
    loginUser(this.state);
  };

  doNothing = () => ({ message: 'Processing Request' });

  handleErrorMessages = () => {
    const { errorMessages } = this.props;
    return errorMessages.errors
      ? {
        nameError: errorMessages.errors.name ? errorMessages.errors.name[0] : '',
        passwordError: errorMessages.errors.password ? errorMessages.errors.password[0] : '',
        generalError: errorMessages.errors.general || ''
      }
      : {
        nameError: '',
        passwordError: '',
        generalError: ''
      };
  };

  modalName = () => {
    const { resetPasswordModal } = this.state;
    if (resetPasswordModal) {
      return 'Forgot Password';
    }
    return 'Login';
  }

  submitReset = async (e) => {
    this.setState({ fetching: true });
    e.preventDefault();
    const { email } = this.state;
    try {
      const response = await post(`${BASE_URL}/users/reset-password`, {
        email
      });
      const { message } = response.data;
      this.setState({ message, fetching: false });
    } catch (error) {
      const { message } = error.response.data;
      if (message === 'user not found in our records') {
        this.setState({
          message: 'Please, verify password reset link in your email box',
          fetching: false
        });
      }
    }
  }

  /**
   * @description social Login
   * @param {string} URL
   * @returns {null} null
   */
  socialMediaLogin = (URL) => {
    window.location.assign(URL);
  };

  /**
   * @returns {HTMLElement} div
   */
  render() {
    const {
      state, doNothing, handleSubmit, handleInputChange, handleErrorMessages, modalName, submitReset
    } = this;
    const {
      rememberMe,
      resetPasswordModal,
      message,
      fetching
    } = state;
    const {
      onClose,
      isLoggedIn,
      loading,
      toggleSignUpModal
    } = this.props;
    const { nameError, passwordError, generalError } = handleErrorMessages();
    return isLoggedIn ? (
      <Redirect to="/profile" />
    ) : (
      <Modal customClass="login_body_modal" modalHeader={modalName()} onClose={onClose}>
        { resetPasswordModal && (
          <div className="login_body">
            <p className="login_body_required_text">*(Required fields)</p>
            <p className="login_body_instruction">PLEASE ENTER YOUR REGISTERED EMAIL ADDRESS</p>
            <section className="login_body_reset_section">
              <form onSubmit={submitReset}>
                <InputField
                  placeHolder="Enter Email"
                  showRequiredAsterisk
                  required
                  inputType="email"
                  customClass="login_body_reset_input"
                  inputName="email"
                  onChange={handleInputChange}
                />
                <br />
                <p className="login_body_required_text login_body_response_text">{ message }</p>
                <Button
                  customClass="login_body_form_submit_btn login_body_reset_btn"
                  btnText="Recover"
                  btnType="submit"
                  isDisabled={fetching}
                />
              </form>
            </section>
          </div>
        )}
        { !resetPasswordModal && (
          <div className="login_body">
            <p className="login_body_required_text">*(Required fields)</p>
            <section className="login_body_info">
              <section className="login_body_socials">
                <Button
                  btnText="Sign in with Facebook"
                  customClass="login_body_socials_btn login_body_socials_facebook facebookDesktop"
                  imgSrc={faceBookLogo}
                  imgAltText="facebook logo"
                  imgCustomClass="login_body_socials_btn_img"
                  onClick={() => this.socialMediaLogin(FACEBOOK_SOCIAL_LOGIN_URL)}
                />
                <Button
                  btnText="Sign in with Twitter"
                  customClass="login_body_socials_btn login_body_socials_twitter twitterDesktop"
                  imgSrc={twitterLogo}
                  imgAltText="twitter logo"
                  imgCustomClass="login_body_socials_btn_img"
                  onClick={() => this.socialMediaLogin(TWITTER_SOCIAL_LOGIN_URL)}
                />
                <Button
                  btnText="Sign in with Google"
                  customClass="login_body_socials_btn login_body_socials_google googleDesktop"
                  imgSrc={googleLogo}
                  imgAltText="google logo"
                  imgCustomClass="login_body_socials_btn_img"
                  onClick={() => this.socialMediaLogin(GOOGLE_SOCIAL_LOGIN_URL)}
                />
              </section>
              <section className="login_body_socials_responsive">
                <p className="login_body_socials_responsive_text">Sign in with</p>
                <Button
                  btnText=""
                  customClass="login_body_socials_btn login_body_socials_facebook facebookMobile"
                  imgSrc={faceBookLogo}
                  imgAltText="facebook logo"
                  imgCustomClass="login_body_socials_responsive_btn_img"
                  onClick={() => this.socialMediaLogin(FACEBOOK_SOCIAL_LOGIN_URL)}
                />
                <Button
                  btnText=""
                  customClass="login_body_socials_btn login_body_socials_twitter twitterMobile"
                  imgSrc={twitterLogo}
                  imgAltText="twitter logo"
                  imgCustomClass="login_body_socials_responsive_btn_img"
                  onClick={() => this.socialMediaLogin(TWITTER_SOCIAL_LOGIN_URL)}
                />
                <Button
                  btnText=""
                  customClass="login_body_socials_btn login_body_socials_google googleMobile"
                  imgSrc={googleLogo}
                  imgAltText="google logo"
                  imgCustomClass="login_body_socials_responsive_btn_img"
                  onClick={() => this.socialMediaLogin(GOOGLE_SOCIAL_LOGIN_URL)}
                />
              </section>
              <section className="login_body_divider">
                <hr className="login_body_divider_line" />
                <span className="login_body_divider_span">OR</span>
              </section>
              <form
                className="login_body_form"
                id="loginForm"
                onSubmit={loading ? doNothing : handleSubmit}
              >
                {generalError && <p className="login_body_form_error_general">{generalError}</p>}
                <InputField
                  placeHolder="Email or Username"
                  showRequiredAsterisk
                  required
                  inputType="text"
                  customClass="login_body_form_input"
                  inputName="name"
                  onChange={handleInputChange}
                />
                {nameError && (
                  <p className="login_body_form_error login_body_form_error_name">{nameError}</p>
                )}
                <InputField
                  placeHolder="Password"
                  showRequiredAsterisk
                  required
                  inputType="password"
                  customClass="login_body_form_input"
                  inputName="password"
                  onChange={handleInputChange}
                />
                {passwordError && <p className="login_body_form_error">{passwordError}</p>}
                <div className="login_body_form_rem_div">
                  <InputField
                    inputType="checkbox"
                    customClass="login_body_form_rem_div_input"
                    inputName="rememberMe"
                    onChange={() => this.setState({ rememberMe: !rememberMe })}
                  />
                  <p className="login_body_form_rem_div_text">Remember me</p>
                </div>
                <Button
                  customClass="login_body_form_submit_btn"
                  btnText="Login"
                  isDisabled={loading}
                  btnType="submit"
                  onClick={doNothing}
                />
                <div className="login_body_form_links">
                  <button type="button" className="ah_signup_registered-link hasAccount" onClick={toggleSignUpModal}>Register Now</button>
                  <button type="button" className="ah_signup_registered-link hasAccount" onClick={() => this.setState({ resetPasswordModal: !resetPasswordModal })}>Forgot Password?</button>
                </div>
              </form>
            </section>
          </div>
        )}
      </Modal>
    );
  }
}

/**
 * @description function to map dispatch to component as props
 * @param {object} dispatch
 * @returns {object} props
 */
export const mapDispatchToProps = dispatch => bindActionCreators(
  {
    loginUser: loginUserAction,
    loadingAuth: loadingAuthAction,
    clearAuthError: clearAuthErrorAction
  },
  dispatch
);

/**
 * @description function to map store to component as props
 * @param {object} store The redux store
 * @returns {object} props
 */
export const mapStateToProps = ({ auth }) => {
  const { isLoggedIn, errorMessages, loading } = auth;
  return {
    isLoggedIn,
    errorMessages,
    loading
  };
};

LoginModal.propTypes = {
  onClose: func.isRequired,
  loginUser: func.isRequired,
  loadingAuth: func.isRequired,
  clearAuthError: func.isRequired,
  loading: bool.isRequired,
  isLoggedIn: bool.isRequired,
  errorMessages: objectOf(object).isRequired,
  toggleSignUpModal: func.isRequired
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(LoginModal);
