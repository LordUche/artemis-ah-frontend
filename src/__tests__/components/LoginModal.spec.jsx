import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, mount } from 'enzyme';
import sinon from 'sinon';
import { LoginModal, mapStateToProps } from '../../views/LoginModal';

const loginModalView = new LoginModal();

describe('Loading Modal Component', () => {
  it('should have match the given snapshot', () => {
    const mockFunction = jest.fn();
    const tree = renderer
      .create(
        <LoginModal
          errorMessages={{}}
          onClose={mockFunction}
          loginUser={mockFunction}
          loadingAuth={mockFunction}
          clearAuthError={mockFunction}
          loading={false}
          isLoggedIn={false}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render modal if user is logged in', () => {
    const mockFunction = jest.fn();
    const loginModal = shallow(
      <LoginModal
        errorMessages={{}}
        onClose={mockFunction}
        loginUser={mockFunction}
        loadingAuth={mockFunction}
        clearAuthError={mockFunction}
        loading={false}
        isLoggedIn={false}
      />
    );
    expect(loginModal.find('Modal').exists()).toBe(true);
    expect(loginModal.find('Redirect').exists()).toBe(false);
  });

  it('should redirect to profile if user is logged in', () => {
    const mockFunction = jest.fn();
    const loginModal = shallow(
      <LoginModal
        errorMessages={{}}
        onClose={mockFunction}
        loginUser={mockFunction}
        loadingAuth={mockFunction}
        clearAuthError={mockFunction}
        loading={false}
        isLoggedIn
      />
    );
    expect(loginModal.find('Redirect').exists()).toBe(true);
  });

  it('should map the right state to props', () => {
    const mockState = {
      auth: {
        isLoggedIn: false,
        errorMessages: [],
        loading: false
      }
    };
    const loginModalProps = mapStateToProps(mockState);
    expect(loginModalProps.isLoggedIn).toEqual(false);
    expect(loginModalProps.errorMessages).toEqual([]);
    expect(loginModalProps.loading).toEqual(false);
  });

  it('should have the right default state', () => {
    const mockFunction = jest.fn();
    const loginModal = shallow(
      <LoginModal
        errorMessages={{}}
        onClose={mockFunction}
        loginUser={mockFunction}
        loadingAuth={mockFunction}
        clearAuthError={mockFunction}
        loading={false}
        isLoggedIn
      />
    );

    expect(loginModal.state('name')).toEqual('');
    expect(loginModal.state('password')).toEqual('');
    expect(loginModal.state('rememberMe')).toEqual(false);
  });

  it('should fire loadingAuth and loginUser function on form submit ', () => {
    const mockLoadingAuth = jest.fn();
    const mockLoginUser = jest.fn();
    const mockFunction = jest.fn();
    const fakeEvent = { preventDefault: () => ({}) };
    const loginModal = mount(
      <LoginModal
        errorMessages={{}}
        onClose={mockFunction}
        loginUser={mockLoginUser}
        loadingAuth={mockLoadingAuth}
        clearAuthError={mockFunction}
        loading={false}
        isLoggedIn={false}
      />
    );

    loginModal.find('form#loginForm').simulate('submit', fakeEvent);

    expect(mockLoadingAuth.mock.calls.length).toEqual(1);
    expect(mockLoginUser.mock.calls.length).toEqual(1);
  });

  it('should clear errors when mounting and when input changes', () => {
    const mockclearAuthError = jest.fn();
    const mockFunction = jest.fn();
    const fakeEvent = { target: { name: 'password', value: 'testValue' } };
    const loginModal = mount(
      <LoginModal
        errorMessages={{
          errors: {
            name: 'Name is required'
          }
        }}
        onClose={mockFunction}
        loginUser={mockFunction}
        loadingAuth={mockFunction}
        clearAuthError={mockclearAuthError}
        loading={false}
        isLoggedIn={false}
      />
    );

    const inputField = loginModal.find('input.login_body_form_input').last();

    inputField.simulate('change', fakeEvent);

    expect(mockclearAuthError.mock.calls.length).toEqual(2);
  });

  it('should update state when input changes', () => {
    const mockclearAuthError = jest.fn();
    const mockFunction = jest.fn();
    const fakeEvent = { target: { name: 'password', value: 'testValue' } };
    const loginModal = mount(
      <LoginModal
        errorMessages={{
          errors: {
            name: 'Name is required',
            password: 'Password is required'
          }
        }}
        onClose={mockFunction}
        loginUser={mockFunction}
        loadingAuth={mockFunction}
        clearAuthError={mockclearAuthError}
        loading={false}
        isLoggedIn={false}
      />
    );

    const passwordInputField = loginModal.find('input.login_body_form_input').last();
    const rememberMeInputField = loginModal.find('input.login_body_form_rem_div_input');

    passwordInputField.simulate('change', fakeEvent);
    expect(loginModal.state('password')).toEqual('testValue');

    rememberMeInputField.simulate('change');
    expect(loginModal.state('rememberMe')).toEqual(true);
  });

  it('should show all errors', () => {
    const mockFunction = jest.fn();
    const loginModal = shallow(
      <LoginModal
        errorMessages={{
          errors: {
            name: ['Name is required'],
            password: ['Password is required'],
            general: 'Invalid Credentials'
          }
        }}
        onClose={mockFunction}
        loginUser={mockFunction}
        loadingAuth={mockFunction}
        clearAuthError={mockFunction}
        loading={false}
        isLoggedIn={false}
      />
    );

    expect(loginModal.find('p.login_body_form_error_general').exists()).toBe(true);
    expect(loginModal.find('p.login_body_form_error_general').text()).toBe('Invalid Credentials');

    expect(
      loginModal
        .find('p.login_body_form_error')
        .first()
        .exists()
    ).toBe(true);
    expect(
      loginModal
        .find('p.login_body_form_error')
        .first()
        .text()
    ).toEqual('Name is required');

    expect(
      loginModal
        .find('p.login_body_form_error')
        .last()
        .exists()
    ).toBe(true);
    expect(
      loginModal
        .find('p.login_body_form_error')
        .last()
        .text()
    ).toEqual('Password is required');
  });

  it('should show only name errors when there are only name errors', () => {
    const mockFunction = jest.fn();
    const loginModal = shallow(
      <LoginModal
        errorMessages={{
          errors: {
            name: ['Name is required']
          }
        }}
        onClose={mockFunction}
        loginUser={mockFunction}
        loadingAuth={mockFunction}
        clearAuthError={mockFunction}
        loading={false}
        isLoggedIn={false}
      />
    );

    expect(loginModal.find('p.login_body_form_error_general').exists()).toBe(false);

    expect(loginModal.find('p.login_body_form_error').exists()).toBe(true);
    expect(loginModal.find('p.login_body_form_error').text()).toEqual('Name is required');
  });

  it('should show only password errors when there are only password errors', () => {
    const mockFunction = jest.fn();
    const loginModal = shallow(
      <LoginModal
        errorMessages={{
          errors: {
            password: ['Password is required']
          }
        }}
        onClose={mockFunction}
        loginUser={mockFunction}
        loadingAuth={mockFunction}
        clearAuthError={mockFunction}
        loading={false}
        isLoggedIn={false}
      />
    );

    expect(loginModal.find('p.login_body_form_error_general').exists()).toBe(false);

    expect(loginModal.find('p.login_body_form_error').exists()).toBe(true);
    expect(loginModal.find('p.login_body_form_error').text()).toEqual('Password is required');
  });

  it('should show only general errors when there are only general errors', () => {
    const mockFunction = jest.fn();
    const loginModal = shallow(
      <LoginModal
        errorMessages={{
          errors: {
            general: 'Invalid Credentials'
          }
        }}
        onClose={mockFunction}
        loginUser={mockFunction}
        loadingAuth={mockFunction}
        clearAuthError={mockFunction}
        loading={false}
        isLoggedIn={false}
      />
    );

    expect(loginModal.find('p.login_body_form_error_general').exists()).toBe(true);
    expect(loginModal.find('p.login_body_form_error_general').text()).toEqual(
      'Invalid Credentials'
    );

    expect(loginModal.find('p.login_body_form_error').exists()).toBe(false);
  });
  it('does not make extra requests when one is being loaded', () => {
    const mockFunction = jest.fn();
    const mockLoadingAuth = jest.fn();
    const mockLoginUser = jest.fn();
    const fakeEvent = { preventDefault: () => ({}) };
    const loginModal = mount(
      <LoginModal
        errorMessages={{}}
        onClose={mockFunction}
        loginUser={mockLoginUser}
        loadingAuth={mockLoadingAuth}
        clearAuthError={mockFunction}
        loading
        isLoggedIn={false}
      />
    );

    loginModal.find('form#loginForm').simulate('submit', fakeEvent);

    expect(mockLoadingAuth.mock.calls.length).toEqual(0);
    expect(mockLoginUser.mock.calls.length).toEqual(0);

    const submitBtn = loginModal.find('button.login_body_form_submit_btn');

    submitBtn.simulate('click');

    expect(mockLoadingAuth.mock.calls.length).toEqual(0);
    expect(mockLoginUser.mock.calls.length).toEqual(0);
  });

  it('Should describe the social media login button click', () => {
    const mockFunction = jest.fn();
    const mockLoadingAuth = jest.fn();
    const mockLoginUser = jest.fn();
    const loginModal = mount(
      <LoginModal
        errorMessages={{}}
        onClose={mockFunction}
        loginUser={mockLoginUser}
        loadingAuth={mockLoadingAuth}
        clearAuthError={mockFunction}
        loading
        isLoggedIn={false}
      />
    );
    sinon.stub(window.location, 'assign');
    loginModal.find('button.facebookDesktop').simulate('click');
    loginModal.find('button.twitterDesktop').simulate('click');
    loginModal.find('button.googleDesktop').simulate('click');
    loginModal.find('button.twitterMobile').simulate('click');
    loginModal.find('button.facebookMobile').simulate('click');
    loginModal.find('button.googleMobile').simulate('click');
  });

  it('Should describe the social media login button click', () => {
    const mockFunction = jest.fn();
    const mockLoadingAuth = jest.fn();
    const mockLoginUser = jest.fn();
    const loginModal = mount(
      <LoginModal
        errorMessages={{}}
        onClose={mockFunction}
        loginUser={mockLoginUser}
        loadingAuth={mockLoadingAuth}
        clearAuthError={mockFunction}
        loading
        isLoggedIn={false}
      />
    );
    loginModal.find('#login-reset-btn').simulate('click');
    const event = {
      preventDefault: () => 'prevent default'
    };
    loginModalView.submitReset(event);
  });
});
