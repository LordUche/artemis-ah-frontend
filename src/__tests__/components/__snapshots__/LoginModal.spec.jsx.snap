// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading Modal Component should have match the given snapshot 1`] = `
<div
  className="ah_modal_backdrop_wrapper"
>
  <div
    className="ah_modal_backdrop"
    onClick={[MockFunction]}
    onKeyPress={[MockFunction]}
    role="button"
    tabIndex="0"
  />
  <div
    className="ah_modal login_body_modal"
  >
    <button
      className="ah_modal_close"
      id="ah-modal-close"
      onClick={[MockFunction]}
      type="button"
    >
      Ã—
    </button>
    <header
      className="ah_modal_header"
    >
      <img
        alt="authors haven logo"
        className="ah_modal_header_img"
        src="https://res.cloudinary.com/artemisah/image/upload/v1554320883/authorshaven/AH_logo_white.png"
      />
      <h2
        className="ah_modal_header_text"
      >
        Login
      </h2>
    </header>
    <div
      className="login_body"
    >
      <p
        className="login_body_required_text"
      >
        *(Required fields)
      </p>
      <section
        className="login_body_info"
      >
        <section
          className="login_body_socials"
        >
          <button
            className="ah-btn login_body_socials_btn login_body_socials_facebook facebookDesktop"
            disabled={false}
            id=""
            onClick={[Function]}
            type="text"
          >
            <img
              alt="facebook logo"
              className="login_body_socials_btn_img"
              src={
                Object {
                  "process": [Function],
                }
              }
            />
            Sign in with Facebook
          </button>
          <button
            className="ah-btn login_body_socials_btn login_body_socials_twitter twitterDesktop"
            disabled={false}
            id=""
            onClick={[Function]}
            type="text"
          >
            <img
              alt="twitter logo"
              className="login_body_socials_btn_img"
              src={
                Object {
                  "process": [Function],
                }
              }
            />
            Sign in with Twitter
          </button>
          <button
            className="ah-btn login_body_socials_btn login_body_socials_google googleDesktop"
            disabled={false}
            id=""
            onClick={[Function]}
            type="text"
          >
            <img
              alt="google logo"
              className="login_body_socials_btn_img"
              src={
                Object {
                  "process": [Function],
                }
              }
            />
            Sign in with Google
          </button>
        </section>
        <section
          className="login_body_socials_responsive"
        >
          <p
            className="login_body_socials_responsive_text"
          >
            Sign in with
          </p>
          <button
            className="ah-btn login_body_socials_btn login_body_socials_facebook facebookMobile"
            disabled={false}
            id=""
            onClick={[Function]}
            type="text"
          >
            <img
              alt="facebook logo"
              className="login_body_socials_responsive_btn_img"
              src={
                Object {
                  "process": [Function],
                }
              }
            />
            
          </button>
          <button
            className="ah-btn login_body_socials_btn login_body_socials_twitter twitterMobile"
            disabled={false}
            id=""
            onClick={[Function]}
            type="text"
          >
            <img
              alt="twitter logo"
              className="login_body_socials_responsive_btn_img"
              src={
                Object {
                  "process": [Function],
                }
              }
            />
            
          </button>
          <button
            className="ah-btn login_body_socials_btn login_body_socials_google googleMobile"
            disabled={false}
            id=""
            onClick={[Function]}
            type="text"
          >
            <img
              alt="google logo"
              className="login_body_socials_responsive_btn_img"
              src={
                Object {
                  "process": [Function],
                }
              }
            />
            
          </button>
        </section>
        <section
          className="login_body_divider"
        >
          <hr
            className="login_body_divider_line"
          />
          <span
            className="login_body_divider_span"
          >
            OR
          </span>
        </section>
        <form
          className="login_body_form"
          id="loginForm"
          onSubmit={[Function]}
        >
          
          <span
            className="ah-input-required-asterisk"
          >
            *
          </span>
          <input
            className="ah-input login_body_form_input"
            maxLength=""
            minLength=""
            name="name"
            onChange={[Function]}
            placeholder="Email or Username"
            required={true}
            type="text"
          />
          
          <span
            className="ah-input-required-asterisk"
          >
            *
          </span>
          <input
            className="ah-input login_body_form_input"
            maxLength=""
            minLength=""
            name="password"
            onChange={[Function]}
            placeholder="Password"
            required={true}
            type="password"
          />
          
          <div
            className="login_body_form_rem_div"
          >
            <input
              className="ah-input login_body_form_rem_div_input"
              name="rememberMe"
              onChange={[Function]}
              placeholder=""
              type="checkbox"
            />
            <p
              className="login_body_form_rem_div_text"
            >
              Remember me
            </p>
          </div>
          <button
            className="ah-btn login_body_form_submit_btn"
            disabled={false}
            id=""
            onClick={[Function]}
            type="submit"
          >
            
            Login
          </button>
          <div
            className="login_body_form_links"
          >
            <button
              className="ah-btn ah_signup_registered-link hasAccount link_lookalike"
              disabled={false}
              id="login-register-btn"
              onClick={[Function]}
              type="button"
            >
              
              Register Now
            </button>
            <a
              className="login_body_form_link"
              href="/"
            >
              Forgot password?
            </a>
          </div>
        </form>
      </section>
    </div>
  </div>
</div>
`;
